shared/middleware/files_api.rb

```diff
@@ -487,10 +487,11 @@ def files_put_file(encrypted_channel_id, filename, body)
     end
     new_entry_hash = JSON.parse new_entry_json
     # Replace downcased filename with original filename (to preserve case in the manifest)
-    new_entry_hash['filename'] = filename
+    new_entry_hash['filename'] = CGI.unescape(filename)
```

Comment:
Codecov reports none of this code has been exercised, even though you just wrote a unit test.  I suspected that coverage reporting for shared tests is broken, so I went ahead and pulled the branch to see if I could easily make this test fail by reversing your API changes.

It was harder to break than I expected, so I'm a little concerned that your new behavior isn't covered.

* All of `test_files.rb` still passes if I delete _all but one_ use of `downcase` from the `files_put_file` method (the `downcase` call on line 486 seems to be the one that's required to pass).  `test_case_insensitivity` fails if I remove that last use of `downcase`.

* All of `test_files.rb` still passes if I delete _all_ uses of `CGI.unescape` from the `files_put_file` but use the existing VCR fixtures.  If I run without VCR `test_escaping_insensitivity` still passes if I remove the `CGI.unescape` call at line 507.

So I still think coverage _might_ be broken since I was able to break the tests with some changes to these methods (@pcardune any ideas?), but I also think there are quite a few uncovered branches here that we should revisit.  Let's not block the bugfix going in on getting this figured out, but do it as a follow-up instead.